### PR TITLE
rgw/adminops: add option to provide storageclass adminops user apis

### PR DESCRIPTION
**Pull request by thotz**
In the `radosgw-admin user` command, the default storage class can defined. This capability is missing in the adminops api.

Fixes https://tracker.ceph.com/issues/66439

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by mattbenjamin — APPROVED** [review]
lgtm--thank you!

File: doc/radosgw/adminops.rst [hunk 1]
```diff
@@ -486,6 +486,19 @@ A tenant name may also specified as a part of ``uid``, by following the syntax
...
 
+``default-placement``
+
+:Description: default placement for the user.
+:Type: string
+:Example: default-placement
+:Required: No
+
+``default-storage-class``
+:Description: default storage class for the user, default-placement need to defined for this option
```
Comment (anthonyeleven) [inline doc/radosgw/adminops.rst:497]: must be defined when setting this option

File: doc/radosgw/adminops.rst [hunk 2]
```diff
@@ -486,6 +486,19 @@ A tenant name may also specified as a part of ``uid``, by following the syntax
...
+:Type: string
+:Example: GLACIER
```
Comment (anthonyeleven) [inline doc/radosgw/adminops.rst:499]: Suggest using a different example, say `STANDARD-1A`.  I've found that if I name an RGW storage class `GLACIER`, at last some clients expect different behavior that RGW isn't prepared to do.

File: doc/radosgw/adminops.rst [hunk 3]
```diff
@@ -683,6 +696,19 @@ Request Parameters
...
 
+``default-placement``
+
+:Description: default placement for the user.
+:Type: string
+:Example: default-placement
+:Required: No
+
+``default-storage-class``
+:Description: default storage class for the user, default-placement need to defined for this option
```
Comment (anthonyeleven) [inline doc/radosgw/adminops.rst:707]: see above

**Review by anthonyeleven — DISMISSED** [review]

**Review by anthonyeleven — COMMENTED** [review]
docs lgtm

**Comment by thotz** [comment]
jenkins test make check

**Comment by thotz** [comment]
jenkins test api

**Comment by cbodley** [comment]
passed qa in https://pulpito.ceph.com/cbodley-2024-06-28_14:14:22-rgw-wip-cbodley-testing-distro-default-smithi/

**Pull request merged**